tt_eager/tt_dnn/op_library/bmm/bmm_op.hpp

```diff
@@ -438,11 +438,11 @@ inline void verify_bmm_inputs(const Tensor& input_tensor_a, const Tensor& input_
 }
 
 // TODO: Should we merge this with matmul and expose an option (or infer it from shape) to bcast_batch
-inline Tensor bmm    (const Tensor &input_tensor_a, const Tensor &input_tensor_b, const MemoryConfig& mem_config = operation::DEFAULT_OUTPUT_MEMORY_CONFIG, std::optional<const DeviceComputeKernelConfig> compute_kernel_config = std::nullopt, bool untilize_out = false) {
+inline Tensor bmm    (const Tensor &input_tensor_a, const Tensor &input_tensor_b, uint8_t cq_id = 0, const MemoryConfig& mem_config = operation::DEFAULT_OUTPUT_MEMORY_CONFIG, std::optional<const DeviceComputeKernelConfig> compute_kernel_config = std::nullopt, bool untilize_out = false) {
```

Comment:
Instead of uint8_t can we make this a type instead?  For example...

class QueueId{
  uint8_t id;
  explicit QueueId(uint8_t id) : id(id){}
  operator uint8_t() const {return id; }
};

I want to be able to have this on the constructor of the operation itself but I need to be able to easily detect it behind the scenes.   My intent is to explore using a Monad pattern on top of the async engine and this would make it easier to fit this within our existing api.

Reply:
I think wrapping this in a class is a good idea for readability. Out of curiosity, I'm wondering how making this a class, instead of just a `uint8_t` makes things easier.

Reply:
The idea is that a unique type will make it easier when doing any kind of TMP (if necessary) in the future.   In general we really need to have unique types wherever possible.  For example ideally we separate out the promise features of the Tensor class into a new type called TensorAsync that simply returns a Tensor class.  (we can talk about this later...)